### DISCUSSION TITLE
How to stop and remove a per-user level service when uninstall application using Wix toolset3

### DISCUSSION THREAD

**Discussion by YingChen1996** [wixtoolset/issues#7883] [Questions]
I want to uninstall an application which created a per -user service when installing msi package.
How can I stop and remove service before uninstalling application?

I have tried codes below:
```
  <CustomAction Id="UninstallMyService"
      Directory="DynamicCliDir"
      Execute="deferred"
      Impersonate="no"
      ExeCommand="cmd.exe /c SC DELETE MyService"
      Return="check" />
  <InstallExecuteSequence>
      <Custom Action="installMyService" Before="InstallFinalize">NOT Installed AND NOT WIX_UPGRADE_DETECTED</Custom>
      <Custom Action="UninstallMyService" Before="RemoveFiles">REMOVE="ALL"</Custom>
  </InstallExecuteSequence>
```
But it will raise error "There is a problem with this Windows Install package. A program run as part of the setup did not finish as expected." when I try to uninstall it in control panel.

**Comment by chrpai** [comment]
Per-user services are a new thing and not supported by MSI.  Now I see your using a custom action deferred no impersonate.  That mean it's running in the SYSTEM context.  Try impersonating to run in the user context.  Also consider using the WiX Quiet execute  custom action pattern as you'll get meaningful logging and no flashing cmd window.   There's also a concern that you'll only be able to clean up one user profile.  If your msi is per-machine I'd reconsider using per-user services to be honest.  If it's per-user then it could work out well enough.
